lib/org/openpsa/invoices/handler/invoice/action.php

```diff
@@ -141,8 +143,59 @@ public function _handler_create_pdf_reminder()
         }
     }    
 
-    public function _handler_send_by_mail()
+    private function load_send_mail_controller() : controller
     {
+        $schemadb = schemadb::from_path($this->_config->get('schemadb_send_mail'));
+        $dm = new datamanager($schemadb);
+        
+        $default_subject = $this->_l10n->get('invoice_mail_title_default');
+        $default_message = $this->_l10n->get('invoice_mail_body_default');
+
+        if ($this->invoice->customer) {
+            $customer = org_openpsa_contacts_group_dba::get_cached($this->invoice->customer);
+            $default_message = $customer->get_parameter('org.openpsa.invoices', 'last_invoice_mail_message') ?: $default_message;
+            $default_subject = $customer->get_parameter('org.openpsa.invoices', 'last_invoice_mail_subject') ?: $default_subject;
+        } elseif ($this->invoice->customerContact) {
+            $contact = org_openpsa_contacts_person_dba::get_cached($this->invoice->customerContact);
+            $default_message = $contact->get_parameter('org.openpsa.invoices', 'last_invoice_mail_message') ?: $default_message;
+            $default_subject = $contact->get_parameter('org.openpsa.invoices', 'last_invoice_mail_subject') ?: $default_subject;
```

Comment:
instead of having the  `?: $default_subject;` four times, we could do this further below at set_defaults